File: jogo.py
def contar_vizinhos_vivos(matriz, x, y, N):
    direcoes = [(-1, -1), (-1, 0), (-1, 1),
                (0, -1),           (0, 1),
                (1, -1),  (1, 0),  (1, 1)]
    
    vizinhos_vivos = 0
    for dx, dy in direcoes:
        nx, ny = x + dx, y + dy
        if 0 <= nx < N and 0 <= ny < N and matriz[nx][ny] == 1:
            vizinhos_vivos += 1
    return vizinhos_vivos

def proximo_estado(matriz, N):
    nova_matriz = [[0]*N for _ in range(N)]
    for i in range(N):
        for j in range(N):
            vivos = contar_vizinhos_vivos(matriz, i, j, N)
            if matriz[i][j] == 1:
                nova_matriz[i][j] = 1 if vivos == 2 or vivos == 3 else 0
            else:
                nova_matriz[i][j] = 1 if vivos == 3 else 0
    return nova_matriz

def main():
    N, Q = map(int, input().split())
    matriz = [list(map(int, list(input().strip()))) for _ in range(N)]

    for _ in range(Q):
        matriz = proximo_estado(matriz, N)

    for linha in matriz:
        print(''.join(map(str, linha)))

if __name__ == "__main__":
    main()

File: jogo.java

import java.util.Scanner;

public class  jogo{
    int[] n,q ;
    import java.util.Scanner;

public class JogoDaVida {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int Q = scanner.nextInt();
        scanner.nextLine(); 

        int[][] matriz = new int[N][N];
        
        
        for (int i = 0; i < N; i++) {
            String linha = scanner.nextLine();
            for (int j = 0; j < N; j++) {
                matriz[i][j] = linha.charAt(j) - '0';
            }
        }
        
  
        for (int step = 0; step < Q; step++) {
            matriz = proximoEstado(matriz, N);
        }

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                System.out.print(matriz[i][j]);
            }
            System.out.println();
        }
        
        scanner.close();
    }

    public static int[][] proximoEstado(int[][] matriz, int N) {
        int[][] novaMatriz = new int[N][N];

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                int vizinhosVivos = contarVizinhosVivos(matriz, i, j, N);

                if (matriz[i][j] == 1) {
                    novaMatriz[i][j] = (vizinhosVivos == 2 || vizinhosVivos == 3) ? 1 : 0;
                } else {
                    novaMatriz[i][j] = (vizinhosVivos == 3) ? 1 : 0;
                }
            }
        }
        return novaMatriz;
    }


    public static int contarVizinhosVivos(int[][] matriz, int x, int y, int N) {
        int[] dx = {-1, -1, -1, 0, 0, 1, 1, 1};
        int[] dy = {-1, 0, 1, -1, 1, -1, 0, 1};
        int vizinhosVivos = 0;

        for (int i = 0; i < 8; i++) {
            int nx = x + dx[i];
            int ny = y + dy[i];

            if (nx >= 0 && nx < N && ny >= 0 && ny < N && matriz[nx][ny] == 1) {
                vizinhosVivos++;
            }
        }
        return vizinhosVivos;
    }
}

    
    

}